### DISCUSSION TITLE
SSE Support?

### DISCUSSION THREAD

**Discussion by lrodriguez14** [apache/apisix#6930] [Q&A] [answered]
Does Apache Apisix support SSE(Server Sent Events)?

**Comment by tzssangglass** [comment]
Can you describe it in more detail?

1. links about SSE
2. how you want to support it and what APISIX needs to do

**Reply by lrodriguez14** [reply]
Hi @tzssangglass,

1. Server Sent Events is a technology that enables a client to receive automatic updates like text-based event data from a server via HTTP connection. Links [1](https://en.wikipedia.org/wiki/Server-sent_events) [2](https://www.w3schools.com/html/html5_serversentevents.asp) [3](https://ably.com/topic/server-sent-events)

2. I already have a Server Running with some SSE endpoints and I can registered to them using curl something like `curl -X GET http://localhost:<Server-port>/<sse-endpoint>` and it leaves an open channel where I continuously receive updates from it. I want to do this using an APISIX where it will leave the communication open and can constantly receive messages and send them to the clients connected. The media type for SSE is `text/event-stream`

Thanks!

**Reply by tzssangglass** [reply]
AFAIK, APISIX does not support it yet. This is interesting, though. Can you send an email to dev@apisix.apache.org? Let's discuss this formally on the mailing list.

**Reply by tzssangglass** [reply]
cc @membphis @spacewander

**Reply by spacewander** [reply]
We can accept this feature if someone can submit a PR for it.

**Comment by akalittle** [comment]
Is SSE available right now ?

**Comment by Harpsichord1207** [comment, marked as answer]
Hi everyone, I've found an artilce https://docs.api7.ai/hub/proxy-buffering, it says to use a proxy_buffering plugin to disable buffering and make sse work. But this plugin only available in APISIX Enterprise Edition.

**Reply by lifehacking** [reply]
@Harpsichord1207  you slove it？

**Reply by zhl11b** [reply]
you slove it？

**Reply by spacewander** [reply]
Actually, we can set the `X-Accel-Buffering` response header to control proxy-buffering with little Lua code. FYI:

> Buffering can also be enabled or disabled by passing “yes” or “no” in the “X-Accel-Buffering” response header field.

https://nginx.org/en/docs/http/ngx_http_proxy_module.html

**Reply by zhaoluping577** [reply]
I tried your method and configured the following in the serverless-post-function plugin, but the problem still exists.
is something wrong? Looking forward to your answer.
"plugins":
{
    "serverless-post-function":
    {
        "phase": "header_filter",
        "_meta":
        {
            "disable": false
        },
        "functions":
        [
            "return function(conf, ctx) ngx.header['X-Accel-Buffering']='no' end"
        ]
    }
}
